src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/TransferProviderVacancyToLegalEntityCommandHandler.cs

```diff
@@ -0,0 +1,90 @@
+    {
+        private readonly ILogger<TransferProviderVacancyToLegalEntityCommandHandler> _logger;
+        private readonly IVacancyRepository _vacancyRepository;
+        private readonly IVacancyTransferService _vacancyTransferService;
+        private readonly IVacancyReviewTransferService _vacancyReviewTransferService;
+        private readonly ITimeProvider _timeProvider;
+        private readonly IMessaging _messaging;
+
+        public TransferProviderVacancyToLegalEntityCommandHandler(ILogger<TransferProviderVacancyToLegalEntityCommandHandler> logger,
+                                                            IVacancyRepository vacancyRepository,
+                                                            IVacancyTransferService vacancyTransferService,
+                                                            IVacancyReviewTransferService vacancyReviewTransferService,
+                                                            ITimeProvider timeProvider, IMessaging messaging)
+        {
+            _logger = logger;
+            _vacancyRepository = vacancyRepository;
+            _vacancyTransferService = vacancyTransferService;
+            _vacancyReviewTransferService = vacancyReviewTransferService;
+            _timeProvider = timeProvider;
+            _messaging = messaging;
+        }
+
+        public async Task Handle(TransferVacancyToLegalEntityCommand message, CancellationToken cancellationToken)
+        {
+            var vacancy = await _vacancyRepository.GetVacancyAsync(message.VacancyReference);
+
+            if (vacancy.OwnerType == OwnerType.Provider)
+            {
+                var vacancyUser = new VacancyUser
+                {
+                    UserId = message.UserRef.ToString(),
+                    Email = message.UserEmailAddress,
+                    Name = message.UserName
+                };
+
+                await ProcessTransferringVacancy(vacancy, vacancyUser, message.TransferReason);
+            }
+        }
+
+        private async Task ProcessTransferringVacancy(Vacancy vacancy, VacancyUser user, TransferReason transferReason)
+        {
+            _logger.LogInformation($"Starting transfer of vacancy {vacancy.VacancyReference.Value} to Legal Entity. Transfer reason: {transferReason.ToString()}");
+            var originalStatus = vacancy.Status;
+
+            await _vacancyTransferService.TransferVacancyToLegalEntityAsync(vacancy, user, transferReason);
+
+            await _vacancyRepository.UpdateAsync(vacancy);
+
+            switch (originalStatus)
+            {
+                case VacancyStatus.Submitted:
+                    await _vacancyReviewTransferService.CloseVacancyReview(vacancy.VacancyReference.GetValueOrDefault(), transferReason);
+                    break;
+                case VacancyStatus.Approved:
+                case VacancyStatus.Live:
+                    await _messaging.PublishEvent(new VacancyClosedEvent
```

Comment:
Why not use VacancyServices.CloseVacancyImmediately?

Reply:
Because you said you were going to change the VacancyServices class. Was going to change after to avoid conflicting with it now.